src/Rules.py

```diff
@@ -14,9 +16,33 @@
 if TYPE_CHECKING:
     from . import ManualWorld
 
+class LogicErrorSource(IntEnum):
+    INFIX_TO_POSTFIX = 1 # includes more closing parentheses than opening (but not the opposite)
+    EVALUATE_POSTFIX = 2 # includes missing pipes and missing value on either side of AND/OR
+    EVALUATE_STACK_SIZE = 3 # includes missing curly brackets
+
+def construct_logic_error(location_or_region: dict, source: LogicErrorSource) -> KeyError:
+    object_type = "location/region"
+    object_name = location_or_region.get("name", "Unknown")
+
+    if location_or_region.get("is_region", False) or "starting" in location_or_region or "connects_to" in location_or_region:
+        object_type = "region"
+    elif "region" in location_or_region or "category" in location_or_region:
```

Comment:
Is there a reason this isn't just an `else` statement? Both region and category are optional when defining a location so it is possible for this to break.

Reply:
Never mind I think I worked my way through the logic.

Reply:
Yeah, just coming back to this, but it's basically saying "if we can reasonably tell it's a region, it's a region; else, if we can reasonably tell it's a location, it's a location; else, default to saying it's either from above the if/else".